python/lsst/meas/astrom/astrometry.py

```diff
@@ -358,16 +377,64 @@ def _matchAndFitWcs(self, refCat, sourceCat, goodSourceCat, refFluxField, bbox,
             displayAstrometry(
                 refCat=refCat,
                 sourceCat=matchRes.usableSourceCat,
-                matches=matchRes.matches,
+                matches=matches,
                 exposure=exposure,
                 bbox=bbox,
                 frame=frame + 2,
                 title="Fit TAN-SIP WCS",
             )
 
         return pipeBase.Struct(
-            matches=matchRes.matches,
+            matches=matches,
             wcs=fitWcs,
             scatterOnSky=scatterOnSky,
             match_tolerance=matchRes.match_tolerance,
         )
+
+    def _removeMagnitudeOutliers(self, sourceFluxField, refFluxField, matchesIn):
+        """Remove magnitude outliers, computing a simple zeropoint.
+
+        Parameters
+        ----------
+        sourceFluxField : `str`
+            Field in source catalog for fluxes.
+        refFluxField : `str`
+            Field in reference catalog for fluxes.
+        matchesIn : `list` [`lsst.afw.table.ReferenceMatch`]
+            List of source/reference matches input
+
+        Returns
+        -------
+        matchesOut : `list` [`lsst.afw.table.ReferenceMatch`]
+            List of source/reference matches with magnitude
+            outliers removed.
+        """
+        nMatch = len(matchesIn)
+        sourceMag = np.zeros(nMatch)
+        refMag = np.zeros(nMatch)
+        for i, match in enumerate(matchesIn):
+            sourceMag[i] = -2.5*np.log10(match[1][sourceFluxField])
+            refMag[i] = (match[0][refFluxField]*units.nJy).to_value(units.ABmag)
+
+        deltaMag = refMag - sourceMag
+        # Protect against negative fluxes and nans in the reference catalog.
+        goodDelta, = np.where(np.isfinite(deltaMag))
+        zp = np.median(deltaMag[goodDelta])
+        # Use median absolute deviation (MAD) for zpSigma
+        # Also require a minimum scatter to prevent floating-point errors from
+        # rejecting objects in zero-noise tests.
+        zpSigma = np.clip(1.4826*np.median(np.abs(deltaMag[goodDelta] - zp)), 1e-3, None)
+
+        self.log.debug(f"Rough zeropoint from astrometry matches is {zp:.4f} +/- {zpSigma:.4f}.")
+
+        goodStars, = np.where(np.abs(deltaMag[goodDelta] - zp)
+                              <= self.config.magnitudeOutlierRejectionNSigma*zpSigma)
+        goodStars = goodDelta[goodStars]
+        nOutlier = nMatch - goodStars.size
+        self.log.info(f"Removing {nOutlier} of {nMatch} magnitude outliers.")
```

Comment:
How about: info("Removed %s magnitude outliers out of %s total astrometry matches.", nOutlier, nMatch)?

I think it would also be good to have a lazy-formatted debug-level message that includes some more information about those outliers (say, the min/max sigma of the removed outliers?)

Reply:
Because of the indexing, this can't easily be done with a lazy computation, and I don't think it's necessary to go into some great computations just for a debug statement.

Reply:
What did you think about my reworded log message? As written, I feel like it might be a bit hard to get the meaning in the middle of a log.